### PR TITLE
VirtualMachineSnapshotContent and VolumeSnapshotContent resource

### PR DESCRIPTION
**Pull request by jpeimer**
##### Short description:
Add VirtualMachineSnapshotContent and VolumeSnapshotContent resources

### CODE REVIEW

**Review by rnetser — CHANGES_REQUESTED** [review]
Please follow README on what needs to be included when adding a new resource

**Comment by jpeimer** [comment]
> Please follow README on what needs to be included when adding a new resource
aah, the CONTRIBUTING, thanks, adding what's missing. 

VirtualMachineSnapshotContent - not sure if we should make an __init__ for it, from the [doc](https://docs.openshift.com/container-platform/4.11/virt/virtual_machines/virtual_disks/virt-managing-vm-snapshots.html): _It is created by the VM snapshot controller and contains references to all resources required to restore the VM._
We never create it ourselves. We want only to look at it and collect the data. 

VolumeSnapshotContent [doc](https://kubernetes.io/docs/concepts/storage/volume-snapshots/#volume-snapshot-contents):  _In dynamic provisioning, the snapshot common controller creates VolumeSnapshotContent objects._ 

So we have no intention to create any of these resources ourselves at the moment.

**Comment by jpeimer** [comment]
!verified

**Comment by jpeimer** [comment]
!verified

File: ocp_resources/virtual_machine_snapshot_content.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from ocp_resources.resource import NamespacedResource
+
+
+class VirtualMachineSnapshotContent(NamespacedResource):
```
Comment (myakove) [inline ocp_resources/virtual_machine_snapshot_content.py:4]: Please add to_dict method with the required values to create  VirtualMachineSnapshotContent

File: ocp_resources/volume_snapshot_content.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from ocp_resources.resource import Resource
+
+
+class VolumeSnapshotContent(Resource):
```
Comment (myakove) [inline ocp_resources/volume_snapshot_content.py:4]: Please add to_dict method with the required values to create  VolumeSnapshotContent

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/virtual_machine_snapshot_content.py (hunk 1 above)
Comment (jpeimer) [reply ocp_resources/virtual_machine_snapshot_content.py:4]: VirtualMachineSnapshotContent is created by the VM snapshot controller, we don't create it ourselves.

File: ocp_resources/volume_snapshot_content.py (hunk 2 above)
Comment (jpeimer) [reply ocp_resources/volume_snapshot_content.py:4]: VolumeSnapshotContent is created by the snapshot common controller, we don't create it ourselves.

Comment (rnetser) [reply ocp_resources/volume_snapshot_content.py:4]: according to https://kubernetes.io/docs/concepts/storage/volume-snapshots/#static
`A cluster administrator creates a number of VolumeSnapshotContents`
Please add to_dict

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/volume_snapshot_content.py (hunk 2 above)
Comment (jpeimer) [reply ocp_resources/volume_snapshot_content.py:4]: Right, it's about static provisioning, but we always prefer it to be dynamically provisioned
https://kubernetes.io/docs/concepts/storage/volume-snapshots/#volume-snapshot-contents
`In dynamic provisioning, the snapshot common controller creates VolumeSnapshotContent objects. `

Comment (rnetser) [reply ocp_resources/volume_snapshot_content.py:4]: it does not matter; this is a resource that can be manually created and the code needs to support it. when using it in your implementation, you can select how to use it.

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
Please reopen if needed.

**Pull request closed without merging**